doc/md/fundamentals/2-types/12-advanced-types.md

```diff
@@ -10,73 +10,140 @@ Advanced type features enable more flexible and expressive type definitions, inc
+
+```motoko
+type Point = { x : Int; y : Int };
+
+let p1 : Point = { x = 1; y = 2; z = 3 };
+let p2 : Point = { x = 1; y = 2; z = 4; c = "Red"; };
+
+p1 == p2;  // true (structural equality at type `Point`)
+```
+
 ## Generic types
 
 Generic types are used to define type parameters that work with multiple data types, commonly used in [functions](https://internetcomputer.org/docs/motoko/fundamentals/types/functions), [classes](https://internetcomputer.org/docs/motoko/fundamentals/types/objects-classes), and data structures.
 
-```motoko no-repl
+```motoko
 // Generic function
 func identity<T>(x : T) : T {
   return x;
-}
+};
 
-let num = identity<Nat>(42);  // num is Nat
-let txt = identity<Text>("Hello");  // txt is Text
+identity<Nat>(42);  // num is Nat
 ```
 
 A generic class can store any type while maintaining type safety:
 
-```motoko no-repl
+```motoko
 class Box<T>(value : T) {
-  public func get() : T { value };
-}
+  public func open() : T { value };
+};
 
 let intBox = Box<Nat>(10);
-let textBox = Box<Text>("Hello");
+intBox.open();
 ```
 
 ## Recursive types
 
 Recursive types allow a type to refer to itself, enabling the creation of nested structures while maintaining type safety. The base library utilizes recursive types to define linked lists.
 
 ```motoko no-repl
-public type List<T> = ?(T, List<T>);
+type List = ?(Nat, List);
+```
+
+This defines a recursive type for representing a linked list of natural number. Each list is either:
+
+- `null`, representing the empty list.
+
+- `?(head, tail)`, where `head` is a `Nat` and `tail` is another `List`.
+
+```motoko
+?(1, ?(2, ?(3, null)))  // A list: 1 → 2 → 3
 ```
 
+To generalize this structure and support values of any type, we introduce a parameterized type:
+
+```motoko no-repl
+type List<T> = ?(T, List<T>);
+```
+
+This defines a generic linked list, where `T` can be any type (`Nat`, `Text`, `Blob`, or a custom type).
+
 ### Manually reversing a linked list
 
-Reversing a linked list involves iterating through the list and prepending each element to a new list. This approach demonstrates list traversal and structural mutation without using the built-in `reverse` method.
+Reversing a linked list involves iterating through the list and prepending each element to a new list. This approach demonstrates list traversal without using `List.reverse` library function.
 
-``` motoko no-repl
+Non-paremitized type:
+
+```motoko
+type List = ?(Nat, List);
+
+persistent actor NatReverser {
+
+let numbers : List = ?(1, ?(2, ?(3, null)));
+
+// Reverses List
+func reverseNat(l : List) : List {
+  var current = l;
+  var rev : List = null;
+
+  while (current != null) {
+    switch (current) {
+      case (?(head, tail)) {
+        rev := ?(head, rev);
+        current := tail;
+      };
+      case (null) {};
+    };
+  };
+  rev
+};
+
+reverseNat(numbers); // ?(3, ?(2, ?(1, null)))
+};
```

Comment:
```suggestion
persistent actor NatReverser {

  // Lists of naturals
  type List = ?(Nat, List);

  let numbers : List = ?(1, ?(2, ?(3, null)));

  // Reverses List
  func reverseNat(l : List) : List {
    var current = l;
    var rev : List = null;

    while (current != null) {
      switch (current) {
        case (?(head, tail)) {
          rev := ?(head, rev);
          current := tail;
        };
        case (null) {};
      };
    };
    rev
  };

  reverseNat(numbers); // ?(3, ?(2, ?(1, null)))
};
```

Move type inside and indent. Then it should compile in Ninja.
If you want to display output, add (at top_
```
import Debug "mo:base/Debug"
```
And add bottom do 
```
Debug.print(debug_show reversetNat(numbers));  instead.
```